### DISCUSSION TITLE
The path broadcast is not allowed to be accessed for read operations when trying to mint Basic NFT

### DISCUSSION THREAD

**Discussion by Maakai123** [Cyfrin/foundry-full-course-cu#2714] [Q&A] [answered]
I got the below error when trying to mint Basic NFT after deployment

 ```make mint ARGS="--network sepolia"
[⠰] Compiling...
No files changed, compilation skipped
Traces:
  [998738] → new MintBasicNft@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← [Return] 4878 bytes of code

  [3394] MintBasicNft::run()
    ├─ [0] VM::readDir("./broadcast", 3) [staticcall]
    │   └─ ← [Revert] vm.readDir: the path broadcast is not allowed to be accessed for read operations
    └─ ← [Revert] vm.readDir: the path broadcast is not allowed to be accessed for read operations 
Error: 
script failed: vm.readDir: the path broadcast is not allowed to be accessed for read operations  
```

``` My interactions file  // SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Script} from "forge-std/Script.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";
import {BasicNft} from "../src/BasicNft.sol";

//HOW TO MINT
contract MintBasicNft is Script {
    string public constant PUG = 
    "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4jqhy4gkaheg4/?filename=0-PUG.json";

    function run() external {
        address mostRecentlyDeployed = DevOpsTools
        .get_most_recent_deployment("BasicNft", block.chainid);

        mintNftOnContract(mostRecentlyDeployed);
    }


    function mintNftOnContract(address contractAddress) public {
        vm.startBroadcast();
        BasicNft(contractAddress).mintNft(PUG);
        vm.stopBroadcast();
    }
}```

**Comment by EngrPips** [comment, marked as answer]
Hello @Maakai123, Please add the below to your `Foundry.toml` file 

```js
fs_permissions = [{ access = "read", path = "./broadcast" }, {access = "read", path = "./Images/"}]
```

**Reply by Maakai123** [reply]
Ya this works, thanks

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.

**Reply by WuJingweiPayton** [reply]
@EngrPips but why we add ffi=true in the foundry.toml it didn't work?

**Reply by WuJingweiPayton** [reply]
> fs_permissions = [{ access = "read", path = "./broadcast" }, {access = "read", path = "./Images/"}]
it works! it works!

**Reply by EngrPips** [reply]
> > fs_permissions = [{ access = "read", path = "./broadcast" }, {access = "read", path = "./Images/"}]
> > it works! it works!

This worked?

**Reply by EngrPips** [reply]
> > fs_permissions = [{ access = "read", path = "./broadcast" }, {access = "read", path = "./Images/"}]
> > it works! it works!

I believe Foundry changed how the permissions work, `ffi` gives too much access to your machine but this ` fs_permissions = [{ access = "read", path = "./broadcast" }, {access = "read", path = "./Images/"}]` gives Foundry the permission to read the file you need it to read.

**Reply by WuJingweiPayton** [reply]
![image](https://github.com/user-attachments/assets/98f4e7b9-7e52-4b06-9edb-c08eb5a29c81)
@EngrPips  why there is no display a dog in my Metamask?
but when I open the [opensea](https://testnets.opensea.io/assets/sepolia/0xDE426C50c2C7522edda40d6238912AcC96Cd6809/0), there is indeed a dog ~ no happy! I want my dog
![image](https://github.com/user-attachments/assets/0a08081f-ed53-4c3f-95b7-275fbb526d87)

**Reply by EngrPips** [reply]
@WuJingweiPayton Read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2112 It should help you
